### PR TITLE
Don't walk past project root when figuring out exclusion

### PR DESCRIPTION
**Pull request by akx**
Refs https://github.com/charliermarsh/ruff/issues/2034

### CODE REVIEW

File: src/resolver.rs [hunk 1]
```diff
@@ -392,6 +392,11 @@ fn is_file_excluded(
             break;
         }
         let settings = resolver.resolve(path, pyproject_strategy);
+        if !path.starts_with(&settings.project_root) {
+            // Bail out if we've walked past the project root
+            // (so excludes etc. are virtually "rooted" to the project).
+            break;
+        }
```
Comment (charliermarsh) [inline src/resolver.rs:399]: I think something like this could work, but isn't this condition always true?

Comment (charliermarsh) [reply src/resolver.rs:399]: I think we may want this, after the `match`:

```rust
        if path == settings.project_root {
            // Bail out if we've walked past the project root
            // (so excludes etc. are virtually "rooted" to the project).
            break;
        }
```

Comment (akx) [reply src/resolver.rs:399]: Nope, it's not always true – `for path in path.ancestors() {` above :)

Comment (akx) [reply src/resolver.rs]: Yep, you're right :)

Switched to this implementation + added a test that piggybacks on the existing directory structure...

**Pull request merged**